### ISSUE TITLE
Strukturovat větve tak, aby odrážely stavy katalogu

### ISSUE THREAD

**Issue by mbradic** [mbradic/gopas-osnovy#5]
V původním znění od @martinskr :

1) dvelop - ta na ktere oba pracujeme
2)release/1.0 - tak kterou jsme schvalili a zverejnujeme
3) main - ta ktera je aktualne funkcni

**Comment by mbradic** [comment]
@martinskr  3. main - ta která je funkční - co to přesně znamená? Ta která je na webu? Pokud ano, tak tu v repu nemáme :)

**Comment by mbradic** [comment]
@martinskr  Teď mi došlo že vlastně ty moje častý změny jsou v adresáři /cli, takže by mi vlastně ten tvůj systém větví vlastně nevadil, jen bych to rozdělil takhle:

/develop - oba měníme .md fajly kurzů. při release (co to pro nás je?) mergujeme do main.
/develop/cli - já měním kód cli, merguju do /develop kdy se mi zachce

nevadilo by mi ani /develop/{release}, potažmo /develop/{release}/cli, kdybychom věděli, co je to release :)

**Comment by martinskr** [comment]
Ja jsem zvyklej mit to takto:
develop - aktualni pracovni verze (nemusí být funkční)
release/1.0 (funkční release)
release/2.0 (funkční release)
release/3.0 (funkční release)
main (aktiuální funkční verze)

Releasem bych chapal to, co se odpublikovalo do GopasIS.

Ale je mi to jedno. Ja si budu klidne udrzovat tu svou develop a zbytku si nebudu vsimat.

Dík M.

**Comment by mbradic** [comment]
@martinskr  

První úvaha která mě napadla je že bude všechno o hodně jednodušší když si to cli přesunu do jiného repa, takže to udělám :)

Ok, tak jestli to chápu správně, tak by to bylo takto:

1. V okamžiku kdy odevzdáš tuto verzi do Gopas IS, někdo (kdo?) udělá 2 věci:
1.1. Mergne develop/1.0 do main
1.2. Branchne main do nové větve release/1.0

2. Když pak někdo navrhne změnu v katalogu, někdo (kdo?) branchne main do nové větve develop/2.0

Chápu to správně?

**Comment by martinskr** [comment]
Ano. S drobou změnou. Develop by byla jen jedna, tedy zadne verze. Verze by mela release. Prvni verzi jsme udelali v podstate ted, v lete treba udelame zmeny a bude novy release.

**Comment by mbradic** [comment]
@martinskr<https://github.com/martinskr>

1. Větve - ok, zkusíme to dle tvého návrhu, když dám pryč to cli, tak s tím asi nemám problém.1.1. Kdo bude dělat ty branch/merge operace?1.2. Kdy s tim chceš začít? 1.3. Teď mám v issue větvi nachystanou verzi bez cli, kam jí mám mergnout?
2. A tohle je důležitý - znamená to že bychom měli jen 2 releasy do roka?2.1. Přijde mi že bychom měli být pružnější2.2. Co s opravama, překlama a jinýma drobnýma změnama?
    On Tuesday, December 10, 2024 at 04:07:08 PM GMT+1, martinskr ***@***.***> wrote:  
 
 

Ano. S drobou změnou. Develop by byla jen jedna, tedy zadne verze. Verze by mela release. Prvni verzi jsme udelali v podstate ted, v lete treba udelame zmeny a bude novy release.

**Comment by martinskr** [comment]
Já bych pracoval na tý větvi develop, v okamžiku, kdy to odpublikuju, udělám merge do main a do novy vetvě release. Opravy a preklepy bych rešil jen commitama s tím, že dáme prefix:
Add:
Update:
Fix:

Podle je to jednoduchý a na to, co potřebujeme to naprosto stačí, ať si nekomplikjeme zivot.

M.

**Comment by mbradic** [comment]
@martinskr<https://github.com/martinskr>

1. Opravy a překlepy - ok, ale co dál s tím commitem? Bude ležet v develop větvi do dalšího release?2. Jak často navrhuješ dělat release?
A dobrali jsme se pár závěrů - opět - oprav mě jestli se pletu :)
1. Odteď oba pracujeme na develop/1.0 větvi2. Dávám si za úkol ověřit zda teď není v main něco co by chybělo v develop/1.0 a případně to tam mergnu3. Já se dozvím že jsi to poslal do Gopas IS tak, že mi github nahlásí merge do main
Souhlas? :)
Dík
M.
On Tuesday, December 10, 2024 at 04:50:15 PM GMT+1, martinskr ***@***.***> wrote: 
 
 

Já bych pracoval na tý větvi develop, v okamžiku, kdy to odpublikuju, udělám merge do main a do novy vetvě release. Opravy a preklepy bych rešil jen commitama s tím, že dáme prefix:
Add:
Update:
Fix:

Podle je to jednoduchý a na to, co potřebujeme to naprosto stačí, ať si nekomplikjeme zivot.

M.

**Comment by martinskr** [comment]
Souhlas a ano, pokud budou zmeny, mergnu do main a udělám novou větev v release. Bugy nebudou nová verze, to prostě commitnu. Nová verze bude nový kurz nebo nějaká zásadní změna.

Souhlas?

**Comment by mbradic** [comment]
@martinskr<https://github.com/martinskr>

Release je v mých očích cokoliv, co pošleš do Gopas IS. Pokud to nebudeme verzovat, ztratíme se pak při sledování co už je v produkci (na veřejném webu) a co ještě ne.
Takže můj návrh (napiš jestli souhlasíš)
Malé změny (opravy a změny v jednotlivých kurzech které nenutí klienťák něco přeplánovat, by končili v malých releasech (1.1, 1.2, ...)Velké změny (změny napříč kurzy, změny návazností, změny délek kurzů) by končili ve velkých releasech (1.0, 2.0)Velké releasy by byli dvakrát do roka (nejpozději 30.6 a 31.12)
    On Tuesday, December 10, 2024 at 05:53:15 PM GMT+1, martinskr ***@***.***> wrote:  
 
 

Souhlas a ano, pokud budou zmeny, mergnu do main a udělám novou větev v release. Bugy nebudou nová verze, to prostě commitnu. Nová verze bude nový kurz nebo nějaká zásadní změna.

Souhlas?

**Comment by martinskr** [comment]
Míro, do toho se mi nechce.


  1.  Kvůli každému překlepu dělat novou verzi? Vždyť tohle je dohledatelný právě přes popisy těch commitů.
  2.  Nepotřebuješ se vracet k verzi, kde byl překlep. A pokud ano, jde to přes commit.
  3.  Kdyby web importoval .md soubory, mělo by to smysl. Ale ono to tak není. Zadává se to ručně a je to neskutečný voser. Pravděpodobnost lidské chyby a tedy toho, že stejně na webu bude něco jinak je obrovská. Nechci si přidělávat zbytečně práci něčím, co mi nic nepřinese.

Přehled mít budeme. Na webu bude to, co je v main. Historie překlepů bude v Commitech a historie vetsich zmen v releasech.

**Comment by mbradic** [comment]
@martinskr<https://github.com/martinskr>

Možná jen nechápu jak to pro tebe mění oběm práce - jak často zadáváš do Gopas IS změny v osnovách a kolik jich bývá?
A taky moje chápání dosavadního procesu nejspíš nebude dokonalé - zde je můj předpoklad:
1. Lektor X ti nahlási chybějící háček v osnově2. Zadáš to do Gopas IS3. Nikdo to dál nehlídá, jestli se to dostalo na web nebo ne, případně za jak dlouho (pozitivní spin se stejným výsledkem - dál věříme Gopasu že už se o to postará)
Vs potenciální budoucí verze kde
1. Lektor X ti nahlási chybějící háček v osnově2. Zadáš to do Gopas IS3. Založíš novou větev v githubu (uměl bych ti to automatizovat na jeden klik)4. Hlída to automat který umím napsat za odpoledne
Pokud by se ti to i tak nelíbilo, uměl bych ten automat udelat i tak že by to v některých případech hlídalo větve a v některých commity, ale bylo by to složitější

    On Tuesday, December 10, 2024 at 06:35:25 PM GMT+1, martinskr ***@***.***> wrote:  
 
 

Míro, do toho se mi nechce.


 1. Kvůli každému překlepu dělat novou verzi? Vždyť tohle je dohledatelný právě přes popisy těch commitů.
 2. Nepotřebuješ se vracet k verzi, kde byl překlep. A pokud ano, jde to přes commit.
 3. Kdyby web importoval .md soubory, mělo by to smysl. Ale ono to tak není. Zadává se to ručně a je to neskutečný voser. Pravděpodobnost lidské chyby a tedy toho, že stejně na webu bude něco jinak je obrovská. Nechci si přidělávat zbytečně práci něčím, co mi nic nepřinese.

Přehled mít budeme. Na webu bude to, co je v main. Historie překlepů bude v Commitech a historie vetsich zmen v releasech.
